rclpy/src/rclpy/_rclpy_action.c

```diff
@@ -1575,11 +1575,16 @@ rclpy_action_process_cancel_request(PyObject * Py_UNUSED(self), PyObject * args)
     action_server, cancel_request, &cancel_response);
   destroy_cancel_request(cancel_request);
   if (RCL_RET_OK != ret) {
+    const char * original_error = rcl_get_error_string().str;
+    const char * extra_error = "";
     ret = rcl_action_cancel_response_fini(&cancel_response);
+    if (RCL_RET_OK != ret) {
+      extra_error = ".  Also failed to cleanup response.";
```

Comment:
how about the following?

```suggestion
      extra_error = rcl_get_error_string().str;
```

i think the error message should be set in `rcl_action_cancel_response_fini` which possibly only returns "cancel_response argument is null"

Reply:
Ah, this is no good. that is gonna overwrite the message in the same address `gtls_rcutils_error_string`. sorry, ignore me...